ocaml/typing/typecore.ml

```diff
@@ -810,6 +810,27 @@ type partiality_constraint =
   | Check_and_warn_if_partial
   | Check_and_warn_if_total
 
+(* Parsed case rhs's, encoded using Jane_syntax instead of in Parsetree proper.
+   Prefixed by "P" to distinguish from [Typedtree.case_rhs] *)
+type parsed_case_rhs =
+  | Psimple_rhs of Parsetree.expression
+  | Pboolean_guarded_rhs of
+      { guard : Parsetree.expression; rhs: Parsetree.expression }
+  | Ppattern_guarded_rhs of
+      { scrutinee : Parsetree.expression
+      ; cases: Parsetree.case list
+      ; loc: Location.t }
+
+let pc_rhs_of_case case =
```

Comment:
How about renaming to `parsed_case_rhs_of_case`? I don't think we should abbreviate unless the compactness is for some reason clarifying.